services/tasks-svc/internal/bed/bed.go

```diff
@@ -161,6 +160,22 @@ func (ServiceServer) GetBedByPatient(
 		return nil, err
 	}
 
+	// check permissions
+	user := commonPerm.UserFromCtx(ctx)
+	checks := make([]hwauthz.PermissionCheck, 0)
+	checks = append(checks,
+		hwauthz.NewPermissionCheck(user, patientPerm.PatientCanUserGet, patientPerm.Patient(patientId)))
+
+	if result != nil {
+		checks = append(checks,
+			hwauthz.NewPermissionCheck(user, perm.BedCanUserGet, perm.Bed(result.BedID)),
+			hwauthz.NewPermissionCheck(user, roomPerm.RoomCanUserGet, roomPerm.Room(result.RoomID)),
```

Comment:
wouldn't be the check for `RoomCanUserGet` enough, as it implies (through transitivity) that the `bed` can also "be get"?

Reply:
yes, currently that's true. but in the future we maybe have different roles, some may not have access to certain beds, and thus should not see them here